### PR TITLE
add mixup

### PR DESCRIPTION
**Pull request by L1aoXingyu**
这个 PR 增加 mixup/cutmix 数据增强方法，区别于一般的数据增强方法，只需要作用在单个 sample 上，mixup 需要作用在一个 batch 的数据集上。

本来考虑将 mixup_func 和每个 dataset 绑定，不过考虑到多个 dataset 输入的时候，相同的信息需要写多次，而且 concatDataset 会覆盖掉这个信息，最终决定将 mixup_func 和 dataloader 绑定，在 get_batch 中进行 mixup_func 的处理。

### CODE REVIEW

File: configs/common/data/cifar.py [hunk 1]
```diff
@@ -47,8 +48,21 @@
...
     dataset=[
-        LazyCall(CIFAR100Dataset)(root="./", train=True, download=True, transform=train_aug),
+        LazyCall(CIFAR100Dataset)(
+            root="./",
+            train=True,
+            download=True,
+            transform=train_aug,
+        ),
     ],
+    mixup_func=LazyCall(Mixup)(
```
Comment (L1aoXingyu) [inline configs/common/data/cifar.py:58]: 配置 mixup

File: configs/vit_imagenet.py [hunk 2]
```diff
@@ -4,8 +4,12 @@
 from .common.optim import optim
 from .common.data.cifar import dataloader
 
-
 from libai.models import VisionTransformerGraph
+from flowvision.loss.cross_entropy import SoftTargetCrossEntropy
+
+model.num_classes = 100
+model.loss_func = LazyCall(SoftTargetCrossEntropy)()
```
Comment (L1aoXingyu) [inline configs/vit_imagenet.py:11]: mixup 之后，需要使用这个 loss function，所以现在将 loss function 移到了模型的配置里面，方便修改

File: libai/trainer/default.py [hunk 3]
```diff
@@ -370,6 +371,11 @@ def get_batch(cls, data: Instance):
         if isinstance(data, flow.utils.data._utils.worker.ExceptionWrapper):
             data.reraise()
 
+        if mixup_func is not None:
+            images, targets = mixup_func(data.get("images").tensor, data.get("targets").tensor)
+            data.get("images").tensor = images
+            data.get("targets").tensor = targets
```
Comment (L1aoXingyu) [inline libai/trainer/default.py:377]: 获得数据之后，进行 mixup 增强

**Review by rentainhe — APPROVED** [review]

**Review by CPFLAME — APPROVED** [review]
